compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -216,7 +216,10 @@ class Typer extends Namer
         val termName = name.toTermName
 
         def adjustExtension(name: Name) =
-          if required.is(ExtensionMethod) then name.toExtensionName else name
+          if required.is(ExtensionMethod) && termName.endsWith(name.lastPart)
+               // pre-check to avoid forming a new string; form extension only if it has a chance of matching `termName`
```

Comment:
Will `termName.endsWith(name.lastPart)` be always true, given that `termName = name.toTermName`?